x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/helpers.tsx

```diff
@@ -138,3 +138,13 @@ export const setFieldValue = (
       form.setFieldValue(key, val);
     }
   });
+
+export const redirectToDetections = (
+  isSignalIndexExists: boolean | null,
+  isAuthenticated: boolean | null,
+  hasEncryptionKey: boolean | null
+) =>
```

Comment:
nit: I would consider writing TypeScript with explicit return types defined such as this one having `boolean` so later if someone modifies this and accidentally introduces a new return type, it will be caught.

Reply:
The thing here I do not want to say true or false before I get back the answer from the API